### DISCUSSION TITLE
window-save-state default behavior on macOS

### DISCUSSION THREAD

**Discussion by feekApp** [ghostty-org/ghostty#3917] [Q&A]
Hi,
It seems like the `window-save-state = default` (on macOS) does not remember the position and size of a window on `cmd-Q`.

When quitting Ghostty, and restarting, the window is back on the center with the default size. The "default" behavior should be on `cmd-Q`  that the position and size is restored. Similar as now when using `window-save-state = always`. But when using `window-save-state = always` also the latest `pwd` is restored. Similar like f.e. Safari. Restoring `pwd` is not what I want.

So how can I get Ghostty restoring the size and position, but not the `pwd`. Using `working-directory = home` does not work, because that's is overwritten by the `always` option.

Maybe an new option `window-save-state = size-and-position` is a solution?

== Feek

**Comment by ara303** [comment]
:+1: I'd love this. Terminal.app can be resized and moved around, which is ~~all~~ (edit: I was wrong, only position is) retained on quit and reload. So there is native precedent there at least for macOS.

`window-save-state=always` goes too far for the aforementioned reasons. I only want the physicality saved, not its... state. 😅

In terms of potential configuration options:

```
window-save-height-state = true
window-save-width-state = true 
window-save-position-state = true
```

(Could also be expanded to offer control over other state attributes like working directory, tabs/splits, etc.)

#4233 proposes to recall the last location on macOS, but I'm uncertain after reading that if that'd recall location **and** width/height (edit: it doesn't, which is unfortunate). This discussion advocates for recalling both, which is what I want.

As an aside, @feekApp, this discussion was a little tricky to search for. Perhaps renaming to something like "Save width/height state in addition to position on macOS" might be more clear and thus easier to find for upvotes?

**Comment by pluiedev** [comment]
I don't use macOS so I cannot comment on the merit of the idea, but in terms of config design, we would go with a set of flags like `window-save-state = size,position,no-pwd` instead of separate boolean options.

**Reply by ara303** [reply]
For my part, this would be fine. I just felt that when I first started with Ghostty I had one, admittedly very small, moment where I was a little disappointed that `window-save-state` was so limited in the options it allowed. It being "either remember everything or nothing" bothered me a little.

**Comment by taylrfnt** [comment]
Just to add, if you have a terminal size you like/use, you can set the window dimensions using [`window-height` and `window-width`](https://ghostty.org/docs/config/reference#window-height).

I know that doesn't address the question of how these sizes would be retained/persisted if you edit them, but I've found this to pretty much meet my needs.

I'll leave the discussion of the feature implementation & design to the relevant subsystem maintainer(s), but you may want to add "Feature request: " to the title or something similar to flag this as a true feature request

**Reply by ara303** [reply]
That misses the point OP is making here, which I agree with. This discussion proposed is to retain the _the previously last used_ width/height which could be reloaded.
